documentation/API.md

```diff
@@ -124,6 +81,92 @@ Basic login endpoint that will also return (on success):
 }
 ```
 
+#### Get preferences endpoint:
+
+GET: `/get-prefs`
+
+Provides current preferences and language options for the current installation. Called by front-end at the very start of loading in order to configure the environment according to system settings.
+
+#### Get language endpoint:
+
+GET: `/language/<langauge-code>`
+
+Gets the set of localised strings for the core application (i.e. not customisable entities like templates).
+
+#### Verification endpoint
+
+GET: `/verify?uid=<uniqueid>`
+
+Sets the verification record at `<uniqueid>` to `true` and triggers configured Action. Used for verification via email, etc. See [Create Verification Action](List-of-Action-plugins.md) for more info.
+
+#### File download endpoint:
+
+GET: `/file?id=<uniqueId>`
+
+Usage: `GET` request with file database id as a URL query parameter.
+
+This is a public endpoint, but files all have a long uniqueId which should prevent unauthorized access.s
+---
+### Authenticated endpoints
+
+These are all prefixes with `/api`, e.g. `http://localhost:8080/api/login-org`
+
+They require a valid JWT supplied as a Bearer token in the authorization header of the request:
+
+```
+Authorization: Bearer <token>
+```
+
+#### File upload endpoint:
+
+POST: `/upload`
+
+Usage: `POST` request with file(s) in the request `body` form-data:  
+`key: "file" value: <File(s)>`
+
+URL query paramter fields (optional):
+
+- `user_id`
+- `application_serial`
+- `application_response_id`
```

Comment:
How about `template_id` isn't that one of the optional?

Reply:
Yeah, there's actually a few others, have added now: 977b0244